### PR TITLE
feat: Migrate docker preview repo stuff in monorepo

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

Migrate docker preview within the mattermost repo.
This PR will deprecate https://github.com/mattermost/mattermost-docker-preview repository 
Related PR https://github.com/mattermost/delivery-platform/pull/72

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-6655

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/build/docker-preview/config_docker.json [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+        "SessionCacheInMinutes": 10,
+        "SessionIdleTimeoutInMinutes": 0,
+        "WebsocketSecurePort": 443,
+        "WebsocketPort": 80,
+        "WebserverMode": "gzip",
+        "EnableCustomEmoji": false,
+        "EnableEmojiPicker": true,
+        "EnableGifPicker": false,
+        "GfycatApiKey": "2_KtH_W5",
+        "GfycatApiSecret": "3wLVZPiswc3DnaiaFoLkDvB4X0IV6CpMkj4tf2inJRsBY6-FnkT08zGmppWFgeof",
```
Comment (phoinixgrr) [inline server/build/docker-preview/config_docker.json:52]: Irrelevant to the work performed here. 

Have we phased out Gfycat? If so, let's take this opportunity to make the necessary updates accordingly.

Comment (toninis) [reply server/build/docker-preview/config_docker.json:52]: Probably we do not need that yes. @M-ZubairAhmed we can just remove the configuration right ?

Comment (saturninoabril) [reply server/build/docker-preview/config_docker.json:52]: I think in general, this config is outdated and could easily get outdated even if we update now. I wonder if this config should be updated regularly or set to minimum config possible (and let it use the default values if unset). I haven't able to run yet but I'm thinking something like below only. Thoughts @isacikgoz?
```json
{
    "ServiceSettings": {
        "EnableOAuthServiceProvider": true,
        "EnableDeveloper": false,
        "EnableGifPicker": true
    },
    "SqlSettings": {
        "DriverName": "postgres",
        "DataSource": "postgres://mmuser:mostest@localhost/mattermost_test?sslmode=disable\u0026connect_timeout=10\u0026binary_parameters=yes",
        "AtRestEncryptKey": ""
    },
    "FileSettings": {
        "DriverName": "local",
        "Directory": "/mm/mattermost-data",
        "PublicLinkSalt": ""
    },
    "EmailSettings": {
        "SMTPServer": "",
        "SMTPPort": "",
        "PushNotificationContents": "generic"
    },
    "ElasticsearchSettings": {
        "ConnectionUrl": ""
    },
    "PluginSettings": {
        "EnableUploads": true,
        "PluginStates": {}
    }
}
```

**Review by saturninoabril — COMMENTED** [review]
Thanks @toninis! I think this makes sense to move to monorepo but please open a discussion to other teams if they agree with this move.

Overall, looks good but will need to verify especially if the suggestion to set the config to minimum takes effect.

Once moved, the source repository stated at https://hub.docker.com/r/mattermost/mattermost-preview should be updated as well.

**Review by amyblais — APPROVED** [review]

File: server/build/docker-preview/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Mattermost Docker Preview Image
+
+This is a Docker image to install Mattermost in *Preview Mode* for exploring product functionality on a single machine using Docker.
+
+Note: This configuration should not be used in production, as it’s using a known password string and contains other non-production configuration settings, and it does not support upgrade. If you’re looking for a production installation with Docker, please see the [Mattermost Production Docker Deployment Guide](http://docs.mattermost.com/install/prod-docker.html).
+
+To contribute, please see [Contribution Guidelines](https://docs.mattermost.com/developer/contribution-guide.html).
+
+To file issues, [search for existing bugs and file a GitHub issue if your bug is new](https://www.mattermost.org/filing-issues/).
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply server/build/docker-preview/README.md:9]: @saturninoabril - Let's point them to the Dev docs instead: ``https://developers.mattermost.com/contribute/why-contribute/#youve-found-a-bug``

File: server/build/docker-preview/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Mattermost Docker Preview Image
+
+This is a Docker image to install Mattermost in *Preview Mode* for exploring product functionality on a single machine using Docker.
+
+Note: This configuration should not be used in production, as it’s using a known password string and contains other non-production configuration settings, and it does not support upgrade. If you’re looking for a production installation with Docker, please see the [Mattermost Production Docker Deployment Guide](http://docs.mattermost.com/install/prod-docker.html).
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply server/build/docker-preview/README.md:5]: @saturninoabril - Thanks! I agree that the docs link should be updated to Saturn's suggestion.

File: server/build/docker-preview/config_docker.json (hunk 1 above)
Comment (M-ZubairAhmed) [reply server/build/docker-preview/config_docker.json:52]: @toninis yes Gfycat have been completely phased out so we can safely remove these keys

**Review by isacikgoz — APPROVED** [review]

File: server/build/docker-preview/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply server/build/docker-preview/README.md]: Thank you @cwarnermm!

**Review by saturninoabril — APPROVED** [review]
Tested and looks good, thanks @toninis!

**Pull request merged**